src/main/java/com/foundrysoftware/minecraft/plugins/PluginTemplate.java

```diff
@@ -26,12 +51,102 @@ public void onDisable(){
+                                item.setType(Material.TIPPED_ARROW);
+                                item.setAmount(item.getAmount());
+                                PotionMeta potionMeta = (PotionMeta)item.getItemMeta();
+                                potionMeta.setBasePotionData(new PotionData(PotionType.JUMP));
+                                int duration =10;
+                                int amplifier =10;
+                                boolean Visible=true;
+                                potionMeta.addCustomEffect(new PotionEffect(PotionEffectType.CONDUIT_POWER,duration,amplifier),Visible);
+                                item.setItemMeta(potionMeta);
+
+                            }
+                            }
+
+                        }
+                    }else{
+                        if(sender instanceof Player){
+                            Player me =(Player)sender;
+                            PlayerInventory inventory = me.getInventory();
+                            for(int i = 0; i<inventory.getSize(); i++){
+                                ItemStack item = inventory.getItem(i);
+                                if(item == null){
+                                    continue;
+                                }
+                                if(isArrow(item)==true){
+                                    item.setType(Material.TIPPED_ARROW);
+                                    item.setAmount(item.getAmount());
+                                    PotionMeta potionMeta = (PotionMeta)item.getItemMeta();
+                                    potionMeta.setBasePotionData(new PotionData(PotionType.getByEffect(PotionEffectType.getByName(args[0]))));
+                                    int duration =10;
+                                    int amplifier =2;
+                                    boolean Visible=true;
+                                    potionMeta.addCustomEffect(new PotionEffect(PotionEffectType.getByName(args[0]),duration,amplifier),Visible);
+                                    item.setItemMeta(potionMeta);
+                    }
+
+                }
+            }
+        }}}
         return false;
     }
-}
+    //@Override
+    public boolean onShoot(ProjectileHitEvent e){
+       if(e.getEntity() instanceof Arrow) {
+           Arrow arrow = (Arrow) e.getEntity();
+           Entity shooter = (Entity) arrow.getShooter();
+           Location loc;
+           if (e.getHitBlock() != null) {
+               loc = e.getHitBlock().getLocation();
+           } else {
+               loc = e.getHitEntity().getLocation();
+           }
+           Bukkit.getServer().broadcastMessage(loc.toString());
+           if (arrow.hasCustomEffect(PotionEffectType.CONDUIT_POWER)) {
+               loc.getWorld().createExplosion(loc,10,false,false,shooter);//location of explosion,size,setFire,DoBlockDamage,ExplosionOwner
+           }
+           //if(arrow.hasCustomEffect(PotionEffectType.DOLPHINS_GRACE)){
```

Comment:
remove any commented out code, if you need it back you'll be able to get it from git